### ISSUE TITLE
[2.3.1] App hangs in chrome debugger mode if using @react-navigation/drawer

### ISSUE THREAD

**Issue by mjmasn** [software-mansion/react-native-reanimated#2820]
## Description

When using @react-navigation/drawer, the app hangs when entering chrome debugger mode on both platforms. On iOS we've seen an `EXC_BAD_ACCESS` crash with a reanimated related stack (see linked repo for full stack): 

```
com.facebook.react.JavaScript (9)#0	0x0000000190c62d18 in JSC::Heap::releaseDelayedReleasedObjects() ()
#1	0x000000019107f718 in JSC::JSLock::willReleaseLock() ()
#2	0x000000019107f124 in JSC::JSLockHolder::~JSLockHolder() ()
#3	0x000000019076e508 in JSValueProtect ()
#4	0x00000001032f5c04 in facebook::jsc::JSCRuntime::JSCObjectValue::JSCObjectValue(OpaqueJSContext*, std::__1::atomic<bool> const&, OpaqueJSValue*, std::__1::atomic<long>&) at node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp:554
#5	0x00000001032f5c80 in facebook::jsc::JSCRuntime::JSCObjectValue::JSCObjectValue(OpaqueJSContext*, std::__1::atomic<bool> const&, OpaqueJSValue*, std::__1::atomic<long>&) at node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp:553
#6	0x00000001032f5fc4 in facebook::jsc::JSCRuntime::makeObjectValue(OpaqueJSValue*) const at node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp:1358
#7	0x00000001032f5f50 in facebook::jsc::JSCRuntime::cloneObject(facebook::jsi::Runtime::PointerValue const*) at node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp:619
#8	0x00000001032f7fe0 in facebook::jsi::Value::getObject(facebook::jsi::Runtime&) const & at node_modules/react-native/ReactCommon/jsi/jsi/jsi.h:1124
#9	0x0000000103304064 in facebook::jsi::Value::asObject(facebook::jsi::Runtime&) const & at node_modules/react-native/ReactCommon/jsi/jsi/jsi.cpp:299
#10	0x00000001022a3864 in reanimated::ShareableValue::adapt(facebook::jsi::Runtime&, facebook::jsi::Value const&, reanimated::ValueType) at node_modules/react-native-reanimated/Common/cpp/SharedItems/ShareableValue.cpp:61
#11	0x00000001022a541c in reanimated::ShareableValue::adapt(facebook::jsi::Runtime&, facebook::jsi::Value const&, reanimated::RuntimeManager*, reanimated::ValueType) at node_modules/react-native-reanimated/Common/cpp/SharedItems/ShareableValue.cpp:194
```

### Expected behavior

App works normally

### Actual behavior & steps to reproduce

App hangs on a black screen forever, on our production app we get an EXC_BAD_ACCESS crash on iOS that may be related.

## Snack or minimal code example

https://github.com/mjmasn/ReanimatedWithNav6

## Package versions

| name                                 | version |
| ------------------------------------ | ------- |
|@react-native-community/masked-view | ^0.1.11 |
|@react-navigation/bottom-tabs | ^6.0.9 |
|@react-navigation/drawer | ^6.1.8 |
|@react-navigation/native | ^6.0.6 |
|@react-navigation/stack | ^6.0.11 |
|react | 17.0.2 |
|react-native | 0.66.4 |
|react-native-gesture-handler | ^2.1.0 |
|react-native-reanimated | ^2.3.1 |
|react-native-safe-area-context | ^3.3.2 |
|react-native-screens | ^3.10.1 |
| NodeJS                               |     16.10.0    |
| Xcode                                |     13.2    |
| Java                                 |    17     |
| Gradle                               |     7.3.3    |

(Edit: Should add our production app is using react-navigation 5.x but same issues apply)

## Affected platforms

- [x] Android
- [x] iOS
- [ ] Web

**Comment by mjmasn** [comment]
Have done some further investigation today, it looks like the logic for whether to use the legacy implementation of drawer is incorrect in that it always tries to use the legacy implementation because it checks `Animated.isConfigured?.()` which is not a function (unless you change `import Animated from 'react-native-reanimated'` to `import * as Animated from 'react-native-reanimated'` - though this may depend on babel/ts config.

Aside from that it also assumes that Reanimated2 cannot run in chrome debugger so it also switches to legacy implementation in chrome debugger, even if the first bug was fixed.

So TL;DR: Set `useLegacyImplementation={false}` on your Drawer navigator to fix this issue and force the use of Reanimated2.

Side note: Reanimated 1 is no longer supported in Chrome Debugger mode, which I had seen explained on the PR implementing the change but not in the changelog. It's possibly worth calling this out as a breaking change vs 2.2.x as it was working on iOS prior to this.

**Comment by MahmudHasanMenon** [comment]
> Have done some further investigation today, it looks like the logic for whether to use the legacy implementation of drawer is incorrect in that it always tries to use the legacy implementation because it checks `Animated.isConfigured?.()` which is not a function (unless you change `import Animated from 'react-native-reanimated'` to `import * as Animated from 'react-native-reanimated'` - though this may depend on babel/ts config.
> 
> Aside from that it also assumes that Reanimated2 cannot run in chrome debugger so it also switches to legacy implementation in chrome debugger, even if the first bug was fixed.
> 
> So TL;DR: Set `useLegacyImplementation={false}` on your Drawer navigator to fix this issue and force the use of Reanimated2.
> 
> Side note: Reanimated 1 is no longer supported in Chrome Debugger mode, which I had seen explained on the PR implementing the change but not in the changelog. It's possibly worth calling this out as a breaking change vs 2.2.x as it was working on iOS prior to this.

It works. Thanks a lot

**Comment by mjmasn** [comment]
@raarts I think we had the same issue and IIRC it was upgrading all the react-navigation packages to latest that resolved it (mix of v5 and v6 didn't work like the docs said it would). That said the drawer still seems pretty buggy (flickering, positioning issues) but we haven't had chance to look into it properly so may just be to do with our setup.
